### PR TITLE
Integrate llvm-project at https://github.com/llvm/llvm-project/commit/5e9945f624bd

### PR DESCRIPTION
**Pull request by MaheshRavishankar**
* Reset third_party/llvm-project: 5e9945f624bd280f93d0b141ce47d6be0158861a (2022-10-18 16:37:08 +0200): [libc] Fix missing bazel dependency

### CODE REVIEW

**Review by hanhanW — APPROVED** [review]

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/DetachElementwiseFromNamedOps.cpp [hunk 1]
```diff
@@ -44,7 +44,9 @@ struct DetachElementwisePattern
     if (!linalgOp.hasTensorSemantics()) return failure();
 
     // Nothing to do if the output tensor operand is already a fill op.
-    linalg::OpOperandVector outputOperands = linalgOp.getOutputTensorOperands();
+    linalg::OpOperandVector outputOperands = linalgOp.hasBufferSemantics()
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/DetachElementwiseFromNamedOps.cpp:47]: Nit: replace with

```
linalg::OpOperandVector outputOperands;
if (linalgOp.hasBufferSemantics())
  outputOperands = linalgOp.getOutputOperands();
```

File: llvm-external-projects/iree-dialects/lib/Dialect/LinalgTransform/IR/StructuredTransformOpsExt.cpp [hunk 2]
```diff
@@ -128,11 +128,24 @@ static LogicalResult isEquivalentToOpImpl(PatternRewriter &rewriter,
...
-      linalgOp.getOutputs() != linalgModelOp.getOutputs() ||
-      linalgOp.getIndexingMaps() != linalgModelOp.getIndexingMaps() ||
-      linalgOp.getIteratorTypesArray() != linalgModelOp.getIteratorTypesArray())
-    return failure();
+  {
+    linalg::OpOperandVector opInputs = linalgOp.getInputOperands();
+    linalg::OpOperandVector modelInputs = linalgModelOp.getInputOperands();
+    linalg::OpOperandVector opOutputs = linalgOp.getOutputOperands();
+    linalg::OpOperandVector modelOutputs = linalgModelOp.getOutputOperands();
+    auto notEqualFn = [](std::tuple<OpOperand *, OpOperand *> in) -> bool {
```
Comment (MaheshRavishankar) [inline llvm-external-projects/iree-dialects/lib/Dialect/LinalgTransform/IR/StructuredTransformOpsExt.cpp:136]: Nit: I think this function is not needed. Equality of `SmallVector<T>` already compares the values of each elements.

Comment (vmurali) [reply llvm-external-projects/iree-dialects/lib/Dialect/LinalgTransform/IR/StructuredTransformOpsExt.cpp:136]: You should be checking the equality of values, not pointers.

Comment (vmurali) [reply llvm-external-projects/iree-dialects/lib/Dialect/LinalgTransform/IR/StructuredTransformOpsExt.cpp:136]: I guess I am not doing it right now. I should.

**Pull request merged**